R/mappings.R

```diff
@@ -72,7 +72,27 @@ get_mappings <- function(p, local_only = FALSE) {
 
 #' @export
 get_mappings.ggplot <- function(p, local_only = FALSE) {
-  p$mapping
+  global_map <- p$mapping
+
+  if (local_only) {
+    return(global_map)
+  }
+
+  layer_maps <- purrr::map(p$layers, "mapping")
+  layer_names <- purrr::reduce(purrr::map(layer_maps, names), intersect)
```

Comment:
We should probably catch and return early with the global mappings when:

1. No layers
2. No mappings in the layers

Which brings up that we could use some test cases specifically around this feature.